lib/streamlit/caching/memo_decorator.py

```diff
@@ -215,8 +214,8 @@ def __call__(
     ) -> Callable[[F], F]:
         ...
 
-    @staticmethod
     def __call__(
+        self,
```

Comment:
Can you add a comment above the function that says something like:

```
# This should be a static method, but there's a mypy bug that 
# breaks type checking for overloaded static functions: 
# https://github.com/python/mypy/issues/7781
```

So that we remember to not accidentally add the staticmethod annotation back in?